Data Cleaning.Rmd

```diff
@@ -0,0 +1,75 @@
+---
+title: "Data Cleaning"
+author: "Georgia Christodoulou"
+date: "2024-10-17"
+output: html_document
+---
+
+```{r}
+knitr::opts_chunk$set(warning = FALSE, message = FALSE)
+```
+
+```{r}
+library(tidyverse)
+library(dbplyr)
+
+
+application_train <- read.csv("data/application_train.csv")
+application_test <- read.csv("data/application_test.csv")
+
+application_train %>% summary()
+
+```
+
+
+```{r}
+application_cleaning <- function(data) {
+  # remove entire columns
+  data <- data[, !grepl("DOCUMENT|AVG|MODE|MEDI", names(data))]
+
+  # impute NA with 0 for specific columns
+  NA_0 <- grep("SOCIAL|BUREAU", names(data), value = TRUE)
+  for (col in NA_0) {
+    data[is.na(data[[col]]), col] <- 0
+  }
+
+  # impute blanks with NA
+  data[data == ""] <- NA
+
+  # impute NAME_TYPE_SUITE with "Unaccompanied"
+  data$NAME_TYPE_SUITE[is.na(data$NAME_TYPE_SUITE)] <- "Unaccompanied"
+
+  # impute OCCUPATION_TYPE with XNA
+  data$OCCUPATION_TYPE[is.na(data$OCCUPATION_TYPE)] <- "XNA"
+
+  # convert all character columns to factors
+  data[sapply(data, is.character)] <- lapply(data[sapply(data, is.character)], factor)
```

Comment:
One thing we miss here is `TARGET`; this is really a boolean factor, right?

Reply:
Also, there's a few "flag" values that are booleans, like `FLAG_MOBIL`, `FLAG_EMAIL`, etc. I think those we want to trun to a `<logical>` data type

Reply:
I just ran a script looking for any column in `application_train_clean` that has 2 unique values. These are the columns. All these could be booleans.
![columns](https://github.com/user-attachments/assets/fe1a5557-d70b-4e3a-b329-2b5594ea08d8)


The interesting ones are some flags use [0,1] (`LIVE_CITY_NOT_WORK_CITY`, ...) while others use [Y,N] (`FLAG_OWN_CAR`, `FLAG_OWN_REALTY`, ...). This should probably be standardized across all boolean-esque columns.